dev/integration_tests/android_testing/lib/src/constants.dart

```diff
@@ -0,0 +1,162 @@
+
+  /// Matches `AccessibilityAction.ACTION_SCROLL_BACKWARD`.
+  static const AndroidSemanticsAction scrollBackward = AndroidSemanticsAction._(0x2000);
+
+  /// Matches `AccessibilityAction.ACTION_CUT`.
+  static const AndroidSemanticsAction cut = AndroidSemanticsAction._(0x4000);
+
+  /// Matches `AccessibilityAction.ACTION_COPY`.
+  static const AndroidSemanticsAction copy = AndroidSemanticsAction._(0x8000);
+
+  /// Matches `AccessibilityAction.ACTION_PASTE`.
+  static const AndroidSemanticsAction paste = AndroidSemanticsAction._(0x10000);
+
+  /// Matches `AccessibilityAction.ACTION_SET_SELECTION`.
+  static const AndroidSemanticsAction setSelection = AndroidSemanticsAction._(0x20000);
+
+  /// Matches `AccessibilityAction.ACTION_EXPAND`.
+  static const AndroidSemanticsAction expand = AndroidSemanticsAction._(0x40000);
+
+  /// Matches `AccessibilityAction.ACTION_COLLAPSE`.
+  static const AndroidSemanticsAction collapse = AndroidSemanticsAction._(0x80000);
+
+  @override
+  String toString() {
+    switch (id) {
+      case 0x1:
+        return 'AndroidSemanticsAction.focus';
+      case 0x2:
+        return 'AndroidSemanticsAction.clearFoucs';
+      case 0x4:
+        return 'AndroidSemanticsAction.select';
+      case 0x8:
+        return 'AndroidSemanticsAction.clearSelection';
+      case 0x10:
+        return 'AndroidSemanticsAction.click';
+      case 0x20:
+        return 'AndroidSemanticsAction.longClick';
+      case 0x40:
+        return 'AndroidSemanticsAction.accessibilityFocus';
+      case 0x80:
+        return 'AndroidSemanticsAction.clearAccessibilityFocus';
+      case 0x100:
+        return 'AndroidSemanticsAction.nextAtMovementGranularity';
+      case 0x200:
+        return 'AndroidSemanticsAction.previousAtMovementGranularity';
+      case 0x400:
+        return 'AndroidSemanticsAction.nextHtmlElement';
+      case 0x800:
+        return 'AndroidSemanticsAction.previousHtmlElement';
+      case 0x1000:
+        return 'AndroidSemanticsAction.scrollForward';
+      case 0x2000:
+        return 'AndroidSemanticsAction.scrollBackward';
+      case 0x4000:
+        return 'AndroidSemanticsAction.cut';
+      case 0x8000:
+        return 'AndroidSemanticsAction.copy';
+      case 0x10000:
+        return 'AndroidSemanticsAction.paste';
+      case 0x20000:
+        return 'AndroidSemanticsAction.setSelection';
+      case 0x40000:
+        return 'AndroidSemanticsAction.expand';
+      case 0x80000:
+        return 'AndroidSemanticsAction.collapse';
+      default:
+        throw new UnsupportedError('Unknown semantics action: $id');
+    }
+  }
+
+  /// Creates a new [AndroidSemanticsAction] from an integer `value`.
+  ///
+  /// Throws An [UnsupportedError] if the id is not a known Android
+  /// accessibility action.
+  static AndroidSemanticsAction deserialize(int value) {
+    final AndroidSemanticsAction action = new AndroidSemanticsAction._(value);
+    action.toString();
```

Comment:
Is this here to make sure value is valid?
I'd try to be more explicit about the validation here, maybe keep a static int->string map of the values, and here validate that the value is contained in the map, and in toString just query the map?